bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/rest/WebSocketManager.java

```diff
@@ -82,25 +91,49 @@ public String getEvent() {
         }
     }
 
-    public void openSession(@Nullable String sessionToken) throws FreeboxException {
-        WebSocketClient client = new WebSocketClient(apiHandler.getHttpClient());
-        URI uri = getUriBuilder().scheme(getUriBuilder().build().getScheme().contains("s") ? "wss" : "ws").build();
-        ClientUpgradeRequest request = new ClientUpgradeRequest();
-        request.setHeader(ApiHandler.AUTH_HEADER, sessionToken);
-
-        try {
-            client.start();
-            client.connect(this, uri, request);
-        } catch (Exception e) {
-            throw new FreeboxException(e, "Exception connecting websocket client");
+    public void openSession(@Nullable String sessionToken, int reconnectInterval) {
+        if (reconnectInterval > 0) {
+            WebSocketClient client = new WebSocketClient(apiHandler.getHttpClient());
+            URI uri = getUriBuilder().scheme(getUriBuilder().build().getScheme().contains("s") ? "wss" : "ws").build();
+            ClientUpgradeRequest request = new ClientUpgradeRequest();
+            request.setHeader(ApiHandler.AUTH_HEADER, sessionToken);
+
+            stopReconnect();
+            ScheduledExecutorService scheduler = ThreadPoolManager
+                    .getScheduledPool(FreeboxOsBindingConstants.BINDING_ID);
+            reconnectJob = Optional.of(scheduler.scheduleAtFixedRate(() -> {
+                try {
+                    closeSession();
+                    client.start();
+                    client.connect(this, uri, request);
+                    // Update listeners in case we would have lost data while disconnecting / reconnecting
+                    listeners.values().forEach(host -> {
+                        host.handleCommand(new ChannelUID(host.getThing().getUID(), REACHABLE), RefreshType.REFRESH);
+                    });
+                    logger.debug("Websocket manager connected to {}", uri);
+                } catch (Exception e) {
+                    logger.warn("Error connecting websocket client: {}", e.getMessage());
+                }
+            }, 0, reconnectInterval, TimeUnit.MINUTES));
         }
     }
 
-    public void closeSession() {
+    private void stopReconnect() {
+        reconnectJob.ifPresent(job -> job.cancel(true));
+        reconnectJob = Optional.empty();
+    }
+
+    public void dispose() {
+        stopReconnect();
+        closeSession();
+    }
+
+    private void closeSession() {
```

Comment:
Is it normal that there is no call to `client.stop()` anywhere ?

Reply:
Correct, forgot it.